pandas/core/arrays/datetimes.py

```diff
@@ -1949,7 +1949,16 @@ def sequence_to_dt64ns(
             data, inferred_tz = objects_to_datetime64ns(
                 data, dayfirst=dayfirst, yearfirst=yearfirst
             )
-            tz = _maybe_infer_tz(tz, inferred_tz)
+            try:
+                tz = _maybe_infer_tz(tz, inferred_tz)
+            except TypeError as e:
+                #  TODO: make generic conversion instead of only from UTC
```

Comment:
I think you can use `tz_localize_to_utc` to first convert from `inferred_tz` to `UTC` and then `tz_convert_from_utc` as you have it here

Reply:
actually it seemed to work with just `convert_to` using the base int form which seems to be UTC based.
Tests also added.